frappe/__init__.py

```diff
@@ -1490,10 +1490,11 @@ def get_newargs(fn, kwargs):
 	if hasattr(fn, "fnargs"):
 		fnargs = fn.fnargs
 	else:
-		fullargspec = inspect.getfullargspec(fn)
-		fnargs = fullargspec.args
-		fnargs.extend(fullargspec.kwonlyargs)
-		varkw = fullargspec.varkw
+		signature = inspect.signature(fn)
+		fnargs = list(signature.parameters.keys())
+		varkw = "varkw" if "kwargs" in fnargs else None
+		if varkw:
+			fnargs.pop(-1)
```

Comment:
Why was this change made? seems unnecessary and a bit more complicated than the previous block

```suggestion
		signature = inspect.signature(fn)
		fnargs = list(signature.parameters)
		varkw = "kwargs" in fnargs
		if varkw:
			fnargs.pop(-1)
```

Reply:
Replaced `inspect.getfullargspec` with `inspect.signature` to preserve signature of a decorated functions.

<img width="919" alt="Screenshot 2022-04-26 at 17 46 44" src="https://user-images.githubusercontent.com/4463796/165297970-0defe3f4-ec55-4658-9368-888895e86681.png">